rasa/utils/tensorflow/model_data_utils.py

```diff
@@ -64,7 +69,21 @@ def featurize_training_examples(
                 )
         output.append(attribute_to_features)
 
-    return output
+    sparse_feature_sizes = {}
+    if output:
+        sparse_attributes = []
+        for attr, features in output[0].items():
+            if features and features[0].is_sparse():
+                sparse_attributes.append(attr)
+        # we don't support updating label attribute layers at this point
+        if label_attribute in sparse_attributes:
+            sparse_attributes.remove(label_attribute)
+        if training_examples:
+            for attribute in sparse_attributes:
+                sparse_feature_sizes[attribute] = training_examples[
+                    0
+                ].get_sparse_feature_sizes(attribute=attribute, featurizers=featurizers)

```

Comment:
Please refactor this out in a separate function to keep the original function length short and also allow easy testing of this part of code snippet.